Kernel/Storage/StorageDevice.cpp

```diff
@@ -29,9 +29,9 @@ StringView StorageDevice::class_name() const
 
 ErrorOr<size_t> StorageDevice::read(OpenFileDescription&, u64 offset, UserOrKernelBuffer& outbuf, size_t len)
 {
-    u64 index = offset / block_size();
-    size_t whole_blocks = len / block_size();
-    size_t remaining = len % block_size();
+    u64 index = offset >> block_size_log();
+    size_t whole_blocks = len >> block_size_log();
+    size_t remaining = len - (whole_blocks * block_size());
```

Comment:
If we're doing this let's just go the whole nine yards and shift left by `block_size_log()` instead of multiplying.